src/api/services/adminServiceHelpers.ts

```diff
@@ -80,3 +82,58 @@ export const mapClientTypesToAdminEnums = (
     return adminEnum;
   });
 };
+
+export type ApiKeyDTO = {
+  key_id: string;
+  name: string;
+  contact: string;
+  roles: ApiRoleDTO[];
+  created: number;
+  disabled: boolean;
+  service_id: number;
+};
+
+export type ApiKeyAdmin = Omit<ApiKeyDTO, 'roles'> & { roles: string };
+
+function mapAdminRolesToApiRoleDTOs(adminRoles: string[], apiRoleMap: Map<String, ApiRoleDTO>) {
+  return adminRoles.map((adminRole) => {
+    const adminRoleTrimmed = adminRole.trim();
+
+    let apiRole = apiRoleMap.get(adminRoleTrimmed);
+
+    if (apiRole === undefined) {
+      // Add new role to map if not found. Stocks duplicate new objects
```

Comment:
to be a purist, while comments have good intentions, it may be an indication that the code is not very easy to understand and perhaps could be treated in some way (e.g. extract method).

Think about if this logic could be simplified, or if not, maybe we don't need the comments at all.

Worth checking out https://refactoring.guru/smells/comments for this and other code smells!

Reply:
that being said, it's not like all your code is littered with comments - I'm being a bit picky here

Reply:
if you decide to leave it in - what does "Stocks duplicate new objects" mean?

Reply:
Yeah I don't think they're necessary, going to remove them. Was probably just being overly careful. Ahaha that's my attempt at spelling stop... sorry.